doc/reference/kernel/threads/index.rst

```diff
@@ -123,10 +120,79 @@ Once suspended, a thread cannot be scheduled until another thread calls
    a thread since a sleeping thread becomes executable automatically when the
    time limit is reached.
 
+.. _thread_states:
+
+Thread States
+*************
+
+A thread that has no factors that prevent its execution is deemed
+to be **ready**, and is eligible to be selected as the current thread.
+
+A thread that has one or more factors that prevent its execution
+is deemed to be **unready**, and cannot be selected as the current thread.
+
+The following factors make a thread unready:
+
+* The thread has not been started.
+* The thread is waiting for a kernel object to complete an operation.
+  (For example, the thread is taking a semaphore that is unavailable.)
+* The thread is waiting for a timeout to occur.
+* The thread has been suspended.
+* The thread has terminated or aborted.
+
+
+  .. image:: thread_states.svg
+     :align: center
+
+.. _thread_priorities:
+
+Thread Priorities
+******************
+
+A thread's priority is an integer value, and can be either negative or
+non-negative.
+Numerically lower priorities takes precedence over numerically higher values.
+For example, the scheduler gives thread A of priority 4 *higher* priority
+over thread B of priority 7; likewise thread C of priority -2 has higher
+priority than both thread A and thread B.
+
+The scheduler distinguishes between two classes of threads,
+based on each thread's priority.
+
+* A :dfn:`cooperative thread` has a negative priority value.
+  Once it becomes the current thread, a cooperative thread remains
+  the current thread until it performs an action that makes it unready.
+
+* A :dfn:`preemptible thread` has a positive priority value.
```

Comment:
Sorry about that change, since zero is neither positive or negative, what you had previously was correct.
```
* A :dfn:`preemptible thread` has a non-negative priority value.
```
or you could write:
```
* A :dfn:`preemptible thread` has a zero or positive priority value.
```